recipes/cccl/all/conanfile.py

```diff
@@ -10,6 +13,7 @@ class CcclConan(ConanFile):
     homepage = "https://github.com/swig/cccl/"
     url = "https://github.com/conan-io/conan-center-index"
     license = "GPL-3.0-or-later"
+    settings = "compiler"
```

Comment:
```suggestion
    settings = "os", "arch", "compiler", "build_type"
```
Keep settings always full, then, manage the package ID directly in package_id(self). It's a pattern well used in CCI.